src/schema/matchmaking/cancelled.ts

```diff
@@ -0,0 +1,15 @@
+import { Type } from "@sinclair/typebox";
+
+import { defineEndpoint } from "@/generator-helpers.js";
+
+export default defineEndpoint({
+    source: "server",
+    target: "user",
+    description:
+        "Server may send this event at any point when the user is queuing to indicate that the user has been booted out the matchmaking system.",
+    event: {
+        data: Type.Object({
+            reason: Type.String(),
```

Comment:
I'd like to try and keep properties like this as explicit as possible where we can, unless it really doesn't make sense. For example, here I would change `Type.String()` to `Type.Union([ Type.Literal("intentional"), Type.Literal("party_user_left"), Type.Literal("server_playlists_changed") ])`. If we need to add or remove reasons later we can easily do so, but this approach improves the typing for protocol consumers and makes it easy for client devs to see what possible reasons to expect here.

Reply:
And for any free form text there can be additional details field as in responses

Reply:
Amended the commit to hardcode some reasons.